src/main/java/org/truffleruby/options/VerbosityOptionDescription.java

```diff
@@ -59,13 +71,20 @@ public Verbosity checkValue(Object value) {
                 case "2":
                     return Verbosity.TRUE;
                 default:
-                    throw new OptionTypeException(getName(), value.toString());
+                    throw new IllegalArgumentException();
             }
         } else if (value instanceof Verbosity) {
             return (Verbosity) value;
         } else {
-            throw new OptionTypeException(getName(), value.toString());
+            throw new IllegalArgumentException();
         }
     }
 
+    private static final OptionType<Verbosity> OPTION_TYPE = new OptionType<>("Verbosity", Verbosity.FALSE, (Function<String, Verbosity>) text -> checkValueInner(text));
```

Comment:
This cast can probably also be removed if a String -> Verbosity parsing method is extracted.